src/wolfsftp.c

```diff
@@ -4846,7 +4922,80 @@ int SFTP_GetAttributes_Handle(WOLFSSH* ssh, byte* handle, int handleSz,
 
 #elif defined(WOLFSSH_USER_FILESYSTEM)
     /* User-defined I/O support */
-#else
+
+#elif defined(MICROCHIP_MPLAB_HARMONY)
+int SFTP_GetAttributesStat(WS_SFTP_FILEATRB* atr, WSTAT_T* stats)
+{
+    /* file size */
+    atr->flags |= WOLFSSH_FILEATRB_SIZE;
+    atr->sz[0] = (word32)stats->fsize;
+    WLOG(WS_LOG_SFTP, "Setting attrib file size to %u", (unsigned int)stats->fsize);
+    atr->sz[1] = (word32)(0);
+
+    /* file permissions */
+    atr->flags |= WOLFSSH_FILEATRB_PERM;
+    if ((stats->fattrib & SYS_FS_ATTR_DIR) & SYS_FS_ATTR_MASK) {
+        atr->per |= 0x41ED; /* 755 with directory */
+    }
+    else {
+        atr->per |= 0x8000;
+    }
+
+    /* check for read only */
+    if ((stats->fattrib & SYS_FS_ATTR_RDO) & SYS_FS_ATTR_MASK) {
+        atr->per |= 0x124; /* octal 444 */
+    }
+    else {
+        atr->per |= 0x1ED; /* octal 755 */
+    }
+
+    /* volume label */
+    //stats.fattrib == SYS_FS_ATTR_VOL
```

Comment:
Are these TODO or should they be removed?

Reply:
Can be removed. I had gotten all of the file state struct values from documentation and commented out the ones was not using (with a note of SYS_FS_ATTR_VOL in case needing it).